### PR TITLE
SfHero component [WIP]

### PR DESCRIPTION
**Pull request by michasik**
# Related issue
[#149](https://github.com/DivanteLtd/storefront-ui/issues/149)

# Scope of work
I have started working on Hero Component. For now idea is to use SfCarousel and SfBanner components, to create SfHero component. 

# Screenshots of visual changes

# Checklist

- [ ] I followed [composition rules](https://github.com/DivanteLtd/storefront-ui/blob/master/docs/component-rules.md) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/organisms/SfHero/SfHero.vue [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<script src="./SfHero.js"></script>
+<template lang="html" src="./SfHero.html"></template>
+<style lang="scss">
+@import "../../../../../shared/styles/components/SfHero.scss"
```
[the remark this answers is not in the document]
Comment (michasik) [reply packages/vue/src/components/organisms/SfHero/SfHero.vue]: The reason I've chosen to use `SfCarousel` is to follow the principles of atom design, so that organisms are built from molecules and atoms. Correct me if I'm wrong, but here `SfCarousel` is similar to designed `SfHero` - basically it is a carousel with only one visible picture and the pagination. 

I am not gonna argue that it is the best path, because I had a hard time to decide how to approach it myself.

Basically I wonder if:
1. maybe  `SfCarousel` should be more flexible to used in `SfHero`, or as you suggested create a new organism/molecule `SfHero` "from scratch".
2. `SfArrow` should be extended with styling that allows the `transparent` background.

**Comment by michasik** [comment]
The reason I've chosen to use `SfCarousel` is to follow the principles of atom design, so that organisms are built from molecules and atoms. Correct me if I'm wrong, but here `SfCarousel` is similar to designed `SfHero` - basically it is a carousel with only one visible picture and the pagination.

I am not gonna argue that it is the best path, because I had a hard time to decide how to approach it myself.

Basically I wonder if:
1. maybe `SfCarousel` should be more flexible to used in `SfHero`, or as you suggested create a new organism/molecule `SfHero` "from scratch".
2. `SfArrow` should be extended with styling that allows the transparent background.

**Comment by leomp12** [comment]
@michasik imo `SfHero` is not similar to `SfCarousel`, actually the only similarity between them is that they're swiping elements, but it's handled by glidejs, not by the carousel component itself.
Among the components, the arrows and the container is different, also the hero component has bullets for pagination and the carousel does not.

In addition, by default they should use glidejs in different ways, so they have much more differences than similarities I think.

About `SfArrow`, take a look at the `sf-arrow--long` CSS modifier, it makes the background transparent, I'm just not sure about the hover effect styles, as we don't have it from figma designs.
